cmd/create/create.go

```diff
@@ -102,13 +102,9 @@ func (o *Options) Run(cmd *cobra.Command, args []string, cloud *common.Cloud) er
 		cfg.Spot = common.Spot(common.SpotEnabled)
 	}
 
-	id := common.NewRandomIdentifier()
-
-	if o.Name != "" {
-		id = common.NewIdentifier(o.Name)
-		if identifier, err := common.ParseIdentifier(o.Name); err == nil {
-			id = identifier
-		}
+	id := common.NewRandomIdentifier(o.Name)
+	if identifier, err := common.ParseIdentifier(o.Name); err == nil {
```

Comment:
A tangential question - why is the identifier format (and its parsing) so complicated:
```
	re := regexp.MustCompile(`(?s)^tpi-([a-z0-9]+(?:[a-z0-9-]*[a-z0-9])?)-([a-z0-9]+)-([a-z0-9]+)$`)

	if match := re.FindStringSubmatch(string(identifier)); len(match) > 0 && hash(match[1]+match[2], shortLength/2) == match[3] {
```

Reply:
See below

Reply:
Thanks for the explanation - much clearer now.